src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs

```diff
@@ -413,37 +413,56 @@ public class HttpProtocolTests : HttpClientTestBase
             });
         }
 
+        public static IEnumerable<string> GetInvalidStatusLine()
+        {
+            yield return "HTTP/1.1 2345";
+            yield return "HTTP/A.1 200 OK";
+            yield return "HTTP/X.Y.Z 200 OK";
+            yield return "HTTP/1.A 200 OK";
+            yield return "HTTP/1.1 ";
+            yield return "HTTP/1.1 !11";
+            yield return "HTTP/1.1 a11";
+            yield return "HTTP/1.1 abc";
+            yield return "HTTP/1.1\t\t";
+            yield return "HTTP/1.1\t";
+            yield return "HTTP/1.1  ";
+            
+            // Only pass on .NET Core Windows & ManagedHandler.
+            if (PlatformDetection.IsNetCore && PlatformDetection.IsWindows)
+            {
+                yield return "HTTP/0.1 200 OK";
+                yield return "HTTP/3.5 200 OK";
+                yield return "HTTP/1.12 200 OK";
+                yield return "HTTP/12.1 200 OK";
+            }
+            
+            // Only fail on WinRTHandler.
```

Comment:
Can we change the wording here?  It confusing to say "fail". When negatives are used in these 'if' statements, it's basically saying to "skip the test cases on something".

So, the comment would be better if it said:  "Skip these test cases on UAP since the behavior is different".

This comment applies to all comments below which say "Fail" check for the negative, i.e. "!" in the 'if' clause.

Reply:
Also, the word "WinRTHandler" is confusing. We don't have a class with that name.  While it's true that the UAP implementation uses WinRT APIs, that's just an implementation detail.  So, using the term "UAP" is better.

Reply:
Will fix.